src/search.rs

```diff
@@ -0,0 +1,146 @@
+use crate::search_error::SearchError;
```

Comment:
You can add `search_error::SearchError` to `common.rs`, and do `use crate::common::*:`